genie-proto/src/main/proto/genie.proto

```diff
@@ -304,3 +304,14 @@ message AgentHeartBeat {
 
 message ServerHeartBeat {
 }
+
+service JobTerminationService {
+    rpc registerForKillNotification (JobKillListenerRequest) returns (JobKillListenerResponse);
+}
+
+message JobKillListenerRequest {
+    string job_id = 1;
+}
+
+message JobKillListenerResponse {
+}
```

Comment:
How about `JobKillRegistrationRequest\Response` as these are request\response to register for a job kill notification

Reply:
Which name? Listener?
Nothing else in the codebase is named Listener so don't tese messages should be named like this.
Same for Termination and Notification. They're not consistent with the rest of the naming in Java.

Reply:
I meant sticking to the name - `JobKillRegistrationRequest\Response`